client/webserver/site/src/js/orderbook.ts

```diff
@@ -23,6 +23,7 @@ export default class OrderBook {
 
   /* add adds an order to the order book. */
   add (ord: MiniOrder) {
+    if (ord.qtyAtomic === 0) return
```

Comment:
I say let's go with it, but in the name of figuring this out, how bout a console log with info about the order to help figure out why it's here?

Reply:
There are a lot of these.

Reply:
You mean it happens frequently or there are a lot of places in the code?

Reply:
They are orders that are booked, match partially (generating a `update_remaining`), then finish matching in the same epoch. By the time the `update_remaining` notification is generated by `BookRouter`, the order's `Remaining` value is `0`. It's of little consequence in most places because an `unbook_order` notification is sent immediately after. I've put a little cache with a timer in place to make sure that the `unbook_order` comes through for every zero-qty order within a second or so, and they all do. But still for some reason, if not ignored, a zero-qty order will end up persisting on the book somehow, screwing up the chart.

Reply:
I was trying to repro with something very similar.  Two lot order on book, matches with 1 lot taker, hitting `handleUpdateRemainingRoute` (which could use an `as RemainderUpdate` I think), and following with a cancel *in the same epoch* hitting `handleUnbookOrderRoute`, so very similar to what you described.  Still couldn't make this `ord.qtyAtomic` condition hit in this `add` function.  :man_shrugging: